source/boundary_temperature/dynamic_core.cc

```diff
@@ -0,0 +1,993 @@
+    {
+      return get_solidus(get_X(r),get_Pressure(r));
+    }
+
+
+    template <int dim>
+    double
+    DynamicCore<dim>::get_dT(double r) const
+    {
+      return get_T(get_Tc(r),r) - get_Ts(r);
+    }
+
+    template <int dim>
+    void
+    DynamicCore<dim>::update_core_data()
+    {
+      get_specific_heating(core_data.Ti,core_data.Qs,core_data.Es);
+      get_radio_heating(core_data.Ti,core_data.Qr,core_data.Er);
+      get_gravity_heating(core_data.Ti,core_data.Ri,core_data.Xi,core_data.Qg,core_data.Eg);
+      get_adiabatic_heating(core_data.Ti,core_data.Ek,core_data.Qk);
+      get_latent_heating(core_data.Ti,core_data.Ri,core_data.El,core_data.Ql);
+      get_heat_solution(core_data.Ti,core_data.Ri,core_data.Xi,core_data.Eh);
+    }
+
+    template <int dim>
+    struct internal::CoreData
+  DynamicCore<dim>::get_core_data() const
+    {
+      return core_data;
+    }
+
+    template <int dim>
+    double
+    DynamicCore<dim>::get_solidus(double X,double p) const
+    {
+      if (use_bw11)
+        {
+          // Change from weight percent to mole percent.
+          double x,x0=32./88.;
+          if (X<x0)
+            x=56.*X/(32.*(1.-X));
+          else
+            x=1.;
+          // Change from Pa to GPa
+          p*=1e-9;
+          // Fe-FeS system solidus by Buono & Walker (2011)
+          return (-2.4724*p*p*p*p + 28.025*p*p*p + 9.1404*p*p + 581.71*p + 3394.8) * x*x*x*x
+                 +( 1.7978*p*p*p*p - 6.7881*p*p*p - 197.69*p*p - 271.69*p - 8219.5) * x*x*x
+                 +(-0.1702*p*p*p*p - 9.3959*p*p*p + 163.53*p*p - 319.35*p + 5698.6) * x*x
+                 +(-0.2308*p*p*p*p + 7.1000*p*p*p - 64.118*p*p + 105.98*p - 1621.9) * x
+                 +( 0.2302*p*p*p*p - 5.3688*p*p*p + 38.124*p*p - 46.681*p + 1813.8);
+
+        }
+      else
+        {
+          if (composition_dependency)
+            return (Tm0*(1-Theta*X)*(1+Tm1*p+Tm2*pow(p,2)));
+          else
+            return (Tm0*(1-Theta)*(1+Tm1*p+Tm2*pow(p,2)));
+        }
+    }
+
+    template <int dim>
+    double
+    DynamicCore<dim>::get_X(double r) const
+    {
+      double xi_3=pow(r/Rc,3);
+      return X_init/(1-xi_3+Delta*xi_3);
+    }
+
+    template <int dim>
+    void
+    DynamicCore<dim>::update()
+    {
+      core_data.dt    = this->get_timestep();
+      core_data.H     = get_radioheating_rate();
+
+      // It's a bit tricky here.
+      // Didn't use the initialize() function instead because the postprocess is initialized after boundary temperature.
+      // It is not available at the time initialize() function of boundary temperature is called.
+      if (is_first_call==true)
+        {
+          const Postprocess::CoreStatistics<dim> *core_statistics
+            = this->template find_postprocessor<const Postprocess::CoreStatistics<dim> >();
+          AssertThrow(core_statistics!=NULL,
+                      ExcMessage ("Dynamic core boundary condition has to work with dynamic core statistics postprocessor."));
```

Comment:
nice job figuring out how to get at the postprocessors! nice job also adding the assertion to make sure we get an error if things aren't set up to work!